src/ProductView/ProductView.js

```diff
@@ -15,69 +15,98 @@ class ProductView extends Component {
             filter: [{id: '0', value: '', column: null}],
             lastFilterId: 0,
             product: null,
+            configuration: null,
             eventTypes: null,
             error: null,
             activeEventType: {attributes: []},
-            activeEvents: []
+            activeEvents: [],
+            showAllConfigurations: true,
+            loading: true
         };
         this.nextAttributeId = 1;
         //Function binding
         this.onInputChange = this.onInputChange.bind(this);
         this.handleProductData = this.handleProductData.bind(this);
+        this.handleConfigurationData = this.handleConfigurationData.bind(this);
         this.handleEventData = this.handleEventData.bind(this);
         this.handleError = this.handleError.bind(this);
         this.handleEventTypeData = this.handleEventTypeData.bind(this);
-        this.fetchProducts = this.fetchProducts.bind(this);
+        this.fetchProduct = this.fetchProduct.bind(this);
+        this.fetchConfiguration = this.fetchConfiguration.bind(this);
         this.fetchEventTypes = this.fetchEventTypes.bind(this);
         this.fetchEventsFor = this.fetchEventsFor.bind(this);
         this.setActiveEventType = this.setActiveEventType.bind(this);
+        this.setProductConfiguration = this.setProductConfiguration.bind(this);
     }
 
     handleEventData(events) {
-        this.setState({activeEvents: events});
+        this.setState({
+            activeEvents: events
+        });
     }
 
     handleEventTypeData(eventTypes) {
         this.setState({
-            eventTypes: eventTypes
+            eventTypes: eventTypes,
+            loading: false
         });
+        if(eventTypes.length == 0) {
+            this.handleEventData([]);
+            this.setState({
```

Comment:
This has a high potential of failure. Context: setState is registered to be executed at the very end of all function calls. In this method you register 2 times setState. The second call could turn out to overwrite the first one.
Suggestion: Execute setState at the very end of the function.